meta/src/dispatch.rs

```diff
@@ -0,0 +1,98 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::{Data, DeriveInput, Error, Fields, Ident, Variant};
+
+/// Check if a variant has the `#[dispatch(translate_discard)]` attribute
+fn has_translate_discard(variant: &Variant) -> bool {
+    variant.attrs.iter().any(|attr| {
+        attr.path().is_ident("dispatch")
+            && attr
+                .parse_args::<Ident>()
+                .map(|ident| ident == "translate_discard")
+                .unwrap_or(false)
+    })
+}
+
+/// Validate that a variant is a tuple variant with exactly one field
+fn validate_variant(variant: &Variant) -> Result<(), Error> {
+    match &variant.fields {
+        Fields::Unnamed(fields) if fields.unnamed.len() == 1 => Ok(()),
+        _ => Err(Error::new_spanned(
```

Comment:
@lens0021 codeconv complains that we don't test this. But it should be an unreachable code, right?

Alternatively, we could add a unit test for it

Reply:
@terror Hmm one other idea that comes to my mind is to use `compile_fail` in doc tests:

```rs
/// ```compile-fail
/// #[derive(StatementDispatch)]
/// struct Test { }
/// ```
///
/// ```compile_fail
/// #[derive(StatementDispatch)]
/// enum InvalidVariant {
///     Unit,
/// }
/// ```
```

Reply:
Sounds good, I'll add this in!